includes/reader-revenue/my-account/class-woocommerce-my-account.php

```diff
@@ -864,25 +866,60 @@ public static function handle_verify_email_change() {
 			return;
 		}
 		if ( \wp_verify_nonce( $nonce, self::VERIFY_EMAIL_CHANGE_PARAM ) ) {
+			$error     = __( 'Something went wrong.', 'newspack-plugin' );
 			$new_email = \get_user_meta( \get_current_user_id(), self::PENDING_EMAIL_CHANGE_META, true );
 			if ( ! $new_email ) {
-				\wc_add_notice( __( 'Something went wrong.', 'newspack-plugin' ), 'error' );
+				\wc_add_notice( $error, 'error' );
 			} else {
-				\delete_user_meta( \get_current_user_id(), self::PENDING_EMAIL_CHANGE_META );
-				\wp_update_user(
+				$user_id   = \get_current_user_id();
+				$old_email = \wp_get_current_user()->user_email;
+				$update    = \wp_update_user(
 					[
-						'ID'         => \get_current_user_id(),
+						'ID'         => $user_id,
 						'user_email' => $new_email,
 					]
 				);
-				\wc_add_notice( __( 'Your email address has been successfully updated.', 'newspack-plugin' ) );
+				if ( \is_wp_error( $update ) ) {
+					\wc_add_notice( $error, 'error' );
+				} else {
+					\delete_user_meta( \get_current_user_id(), self::PENDING_EMAIL_CHANGE_META );
+					self::sync_email_change( $user_id, $new_email, $old_email );
```

Comment:
`sync_email_change` method might not be successful -> error state should be handled here, instead of assuming success.

Reply:
Yeah. Agreed. [I mentioned in the Asana task](https://app.asana.com/0/0/1209215513701719/1209530560533168/f) that my plan was to tackle handling error cases in a follow-up PR to avoid making these PRs too complex. But if you prefer this be done in a single PR, I can work on that here!